Gems/ONNX/Code/Source/Clients/ONNXSystemComponent.cpp

```diff
@@ -0,0 +1,300 @@
+    {
+        if (ImGui::BeginMenu("ONNX"))
+        {
+            ImGui::MenuItem(m_timingStats.GetName(), "", &m_timingStats.m_show);
+            if (ENABLE_CUDA)
+            {
+                ImGui::MenuItem(m_timingStatsCuda.GetName(), "", &m_timingStatsCuda.m_show);
+            }
+            ImGui::EndMenu();
+        }
+    }
+
+    void ONNXSystemComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<ONNXSystemComponent, AZ::Component>()->Version(0);
+
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<ONNXSystemComponent>("ONNX", "[Description of functionality provided by this System Component]")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("System"))
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true);
+            }
+        }
+    }
+
+    void ONNXSystemComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+    {
+        provided.push_back(AZ_CRC_CE("ONNXService"));
+    }
+
+    void ONNXSystemComponent::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+    {
+        incompatible.push_back(AZ_CRC_CE("ONNXService"));
+    }
+
+    void ONNXSystemComponent::GetRequiredServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+    }
+
+    void ONNXSystemComponent::GetDependentServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& dependent)
+    {
+    }
+
+    ONNXSystemComponent::ONNXSystemComponent()
+    {
+        if (ONNXInterface::Get() == nullptr)
+        {
+            ONNXInterface::Register(this);
+        }
+
+        m_timingStats.SetName("MNIST Timing Statistics");
+        m_timingStats.SetHistogramBinCount(200);
+
+        if (ENABLE_CUDA)
+        {
+            m_timingStatsCuda.SetName("MNIST CUDA Timing Statistics");
+            m_timingStatsCuda.SetHistogramBinCount(200);
+        }
+
+        ImGui::ImGuiUpdateListenerBus::Handler::BusConnect();
+    }
+
+    ONNXSystemComponent::~ONNXSystemComponent()
+    {
+        ImGui::ImGuiUpdateListenerBus::Handler::BusDisconnect();
+
+        if (ONNXInterface::Get() == this)
+        {
+            ONNXInterface::Unregister(this);
+        }
+    }
+
+    Ort::Env* ONNXSystemComponent::GetEnv()
+    {
+        return m_env.get();
+    }
+
+    Ort::AllocatorWithDefaultOptions* ONNXSystemComponent::GetAllocator()
+    {
+        return m_allocator.get();
+    }
+
+    void OnnxLoggingFunction(
+        void*, OrtLoggingLevel, const char* category, const char* logid, const char* code_location, const char* message)
+    {
+        AZ_Printf("\nONNX", "%s %s %s %s", category, logid, code_location, message);
```

Comment:
The category for the Printf shouldn't have a newline in it.

Furhthermore the format string should end with a newline
```suggestion
        AZ_Printf("ONNX", "%s %s %s %s\n", category, logid, code_location, message);
```

Reply:
Fixed for all uses of AZ_Printf